src/Interpreters/Context.cpp

```diff
@@ -274,9 +274,13 @@ struct ContextSharedPart : boost::noncopyable
     ConfigurationPtr users_config;                              /// Config with the users, profiles and quotas sections.
     InterserverIOHandler interserver_io_handler;                /// Handler for interserver communication.
 
+    OnceFlag buffer_flush_schedule_pool_initializer;
```

Comment:
(cosmetic: it would be cool if you could name the `*_initalizer` flags `*_initialized` instead. This is more correct as the variable is essentially an atomic flag and not some "active" function that initializes something)

Reply:
Sure, that makes sense. Updated implementation.